boards/BMS/Inc/App/App_CellMonitors.h

```diff
@@ -0,0 +1,161 @@
+#pragma once
+
+#include "App_SharedExitCode.h"
+
+struct CellMonitors;
+
+enum ITMPInRangeCheck
+{
+    // An indicator for when ITMP is above the threshold to disable the charger
+    // and the threshold to disable cell balancing
+    ITMP_OVERFLOW,
+
+    // An indicator for when ITMP is above the threshold to disable cell
+    // balancing and below the threshold to disable the charger
+    ITMP_CELL_BALANCING_OVERFLOW,
+
+    // An indicator for when ITMP is below the threshold to re-enable the
+    // charger
+    ITMP_CHARGER_IN_RANGE,
+
+    // An indicator for when ITMP is below the threshold to re-enable the
+    // charger and the threshold to re-enable cell balancing
+    ITMP_IN_RANGE
+};
+
+/**
+ * Allocate and initialize a group of cell monitors.
+ * @param read_die_temperatures A function that can be called to read internal
+ * die temperatures (°C) from all cell monitors.
+ * @param get_monitor_0_die_temp A function that returns the internal die
+ * temperature (°C) for the 0th cell monitor.
+ * @param get_monitor_1_die_temp A function that returns the internal die
+ * temperature (°C) for the 1st cell monitor.
+ * @param get_monitor_2_die_temp A function that returns the internal die
+ * temperature (°C) for the 2nd cell monitor.
+ * @param get_monitor_3_die_temp A function that returns the internal die
+ * temperature (°C) for the 3rd cell monitor.
+ * @param get_monitor_4_die_temp A function that returns the internal die
+ * temperature (°C) for the 4th cell monitor.
+ * @param get_monitor_5_die_temp A function that returns the internal die
+ * temperature (°C) for the 5th cell monitor.
+ * @param get_max_die_temp A function that returns the current maximum internal
+ * die temperature (°C) out of all the cell monitors.
+ * @param min_die_temp The minimum die temperature (°C) threshold for the
+ * given cell monitor.
+ * @param max_die_temp The maximum die temperature (°C) threshold for the
+ * given cell monitor.
+ * @param die_temp_re_enable_charger The die temperature (°C) to re-enable the
+ * charger.
+ * @param die_temp_re_enable_cell_balancing The die temperature (°C) to
+ * re-enable cell balancing.
+ * @param die_temp_disable_cell_balancing The die temperature (°C) to disable
+ * cell balancing.
+ * @param die_temp_disable_charger The die temperature (°C) to disable the
+ * charger.
+ * @return A pointer to the created cell monitor whose ownership is given to the
+ * caller.
+ */
+struct CellMonitors *App_CellMonitors_Create(
+    ExitCode (*read_die_temperatures)(void),
+    float (*get_monitor_0_die_temp)(void),
+    float (*get_monitor_1_die_temp)(void),
+    float (*get_monitor_2_die_temp)(void),
+    float (*get_monitor_3_die_temp)(void),
+    float (*get_monitor_4_die_temp)(void),
+    float (*get_monitor_5_die_temp)(void),
+    float (*get_max_die_temp)(void),
+    float min_die_temp,
```

Comment:
@garethellis0 Not sure if you wanted me to get rid of `degc` for the thresholds as well (`max_die_temp, min_die_temp,...`)

Reply:
I think it makes sense to leave `degc` in the name here?